### PR TITLE
rails backend, stage1, with basic tests

### PR DESCRIPTION
**Pull request by meiriko**

### CODE REVIEW

File: app/assets/javascripts/restaurants.coffee [hunk 1]
```diff
@@ -0,0 +1,3 @@
+# Place all the behaviors and hooks related to the matching controller here.
```
Comment (barakcoh) [inline app/assets/javascripts/restaurants.coffee]: you can remove empty files

File: app/assets/javascripts/we_eat.coffee [hunk 2]
```diff
@@ -0,0 +1,3 @@
+# Place all the behaviors and hooks related to the matching controller here.
```
Comment (barakcoh) [inline app/assets/javascripts/we_eat.coffee]: ditto

File: app/assets/stylesheets/restaurants.scss [hunk 3]
```diff
@@ -0,0 +1,3 @@
+// Place all the styles related to the Restaurants controller here.
```
Comment (barakcoh) [inline app/assets/stylesheets/restaurants.scss]: you get the point...

File: app/models/restaurant.rb [hunk 4]
```diff
@@ -0,0 +1,5 @@
+class Restaurant < ApplicationRecord
+    validates :name, :cuisine, :address, presence: true
+    validates :rating, numericality: { only_integer: true, greater_than_or_equal_to: 0, less_than_or_equal_to: 3 }
```
Comment (barakcoh) [inline app/models/restaurant.rb:3]: nice! 👍 
i think rating is 1-5, no?

File: app/models/restaurant.rb [hunk 5]
```diff
@@ -0,0 +1,5 @@
...
+    validates :max_delivery_time, numericality: { only_integer: true, greater_than_or_equal_to: 0 }
```
Comment (barakcoh) [inline app/models/restaurant.rb]: when an `integer` designates time, i like to have the unit included in the variable name.

File: db/migrate/20170831085926_create_restaurants.rb [hunk 6]
```diff
@@ -0,0 +1,14 @@
+class CreateRestaurants < ActiveRecord::Migration[5.1]
+  def change
+    create_table :restaurants do |t|
+      t.string :name
```
Comment (barakcoh) [inline db/migrate/20170831085926_create_restaurants.rb:4]: on top of model-level validations (which you did very well), we also like to enforce basic checks on the db schema level. for instance you can do `t.string :name, null: false`. this prohibits doing damage by bypassing the model checks (e.g. `ActiveRecord::Base.connection.execute('UPDATE restaurants SET name = null')`)

File: test/controllers/restaurants_controller_test.rb [hunk 7]
```diff
@@ -0,0 +1,48 @@
+require 'test_helper'
+
+class RestaurantsControllerTest < ActionDispatch::IntegrationTest
+  setup do
+    @restaurant = restaurants(:one)
+  end
+
+  test "should get index" do
+    get restaurants_url
+    assert_response :success
```
Comment (barakcoh) [inline test/controllers/restaurants_controller_test.rb:10]: while this works, we use `rspec` for testing, i think it's worth knowing at least the basics

File: test/fixtures/restaurants.yml [hunk 8]
```diff
@@ -0,0 +1,17 @@
+# Read about fixtures at http://api.rubyonrails.org/classes/ActiveRecord/FixtureSet.html
+
+one:
```
Comment (barakcoh) [inline test/fixtures/restaurants.yml:3]: we use factories (using `FactoryGirl`) over fixtures as they provide more flexability

File: test/models/restaurant_test.rb [hunk 9]
```diff
@@ -0,0 +1,34 @@
...
+  test "restaurant attributes should not be empty" do
+    restaurant = Restaurant.new
+    assert restaurant.invalid?
+    assert restaurant.errors[:name].any?
+    assert restaurant.errors[:cuisine].any?
+    assert restaurant.errors[:address].any?
+  end
+
+  test "restaurant rating and max delivery time should be in range" do
+    restaurant = Restaurant.new(name: 'rest 1',
```
Comment (barakcoh) [inline test/models/restaurant_test.rb]: same note about factories. for example here, you can have the factory build a `Restaurant` instance using default parameters and only override the one you're specifically interested in, e.g. `rating` without having to supply attributes which are of no interest to the test such as `name`

**Review by barakcoh — APPROVED** [review]
Looks good overall, added some comments about possible improvements

File: app/models/restaurant.rb (hunk 4 above)
Comment (meiriko) [reply app/models/restaurant.rb:3]: from the email: ..... * Its rating (0-3 stars)

**Pull request merged**